### PR TITLE
Add Metallb app for install

### LINKED ISSUE #76: [Feature Request] Add MetalLB
If running on a set of Raspberry Pis or anywhere that does not provide a loadbalancer, we have to fall back to using host ports for Kubernetes. [MetalLB](https://metallb.universe.tf/) solves this issue by providing a bare metal load balancer implementation, and it works quite nicely I would say. I currently use it on all my clusters without any issues, and it would be great to see it added to Arkade.

The issue I see though, is that they have stopped publishing helm charts and moved to only manifests/kustomize as their officially supported deployment methods. 

Please let me know if I can help in any way.

### PR DESCRIPTION
**Pull request by networkop**
## Description
This PR adds support for metallb installation. It installs metallb in [L2 mode](https://metallb.universe.tf/concepts/layer2/) and provides an argument for configurable LB IP range (default: 192.168.0.0/24)

## Motivation and Context
Fixes #76 
- [x] I have raised an issue to propose this change ([required](https://github.com/openfaas/faas/blob/master/CONTRIBUTING.md))

**Notes**:
* It uses the [official](https://metallb.universe.tf/installation/) installation method, it doesn't rely on 3rd party helm charts (e.g. [bitnami](https://bitnami.com/stack/metallb/helm))
* It adds a new function [`KubectlIn`](https://github.com/alexellis/arkade/compare/master...networkop:master#diff-53f55403dea3cbf51eac6f9161c38f774e6e21d86f0d4af6afbb49d9d3468077R68) to allow YAML manifests to be passed via stdin  

## How Has This Been Tested?

Deploy to a new cluster:
```
./arkade install metallb --address-range 172.16.0.0/24
```

Any pending loadBalancer services will soon get a new IP:
```
k get svc -A
NAMESPACE              NAME                                 TYPE           CLUSTER-IP       EXTERNAL-IP   PORT(S)                      AGE
default                ingress-nginx-controller             LoadBalancer   10.108.196.250   172.16.0.0   80:30611/TCP,443:32517/TCP   68s
```

## Types of changes
- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)

## Checklist:

- [ ] My code follows the code style of this project.
- [x] My change requires a change to the documentation.
- [x] I have updated the documentation accordingly.
- [x] I've read the [CONTRIBUTION](https://github.com/openfaas/faas/blob/master/CONTRIBUTING.md) guide
- [x] I have signed-off my commits with `git commit -s`
- [ ] I have added tests to cover my changes.
- [x] All new and existing tests passed.

- [x] I have tested this on arm, or have added code to prevent deployment

### CODE REVIEW

**Comment by alexellis** [comment]
Does it make sense to only install in L2 mode?

What other modes are available and would be typically required?

I understand that Kinvolk maintain parts of metallb, so I'm going to ping @vbatts to see if he can shoulder tap someone.

Alex

**Comment by vbatts** [comment]
cc @rata @johananl @russellb

**Comment by networkop** [comment]
The other alternative is the L3/BGP mode which requires a lot more input from a user and at least one BGP speaker, so (based on my understanding) it's not really compatible with arkade's easy to get started approach. 

L2 mode, otoh can work out of the box on any Linux distro with minimum dependencies.

File: cmd/apps/metallb_app.go [hunk 1]
```diff
@@ -0,0 +1,111 @@
...
+			SecretData: []types.SecretsData{{
+				Type:  "string-literal",
+				Key:   "secretkey",
+				Value: b64.StdEncoding.EncodeToString(token),
+			}},
+		}
+
+		err = k8s.CreateSecret(secret)
+		if err != nil {
+			return fmt.Errorf("Create secret error: %+v", err)
```
Comment (alexellis) [inline cmd/apps/metallb_app.go:71]: Go errors should start with a lowercase letter

File: cmd/apps/metallb_app.go [hunk 2]
```diff
@@ -0,0 +1,111 @@
...
+		}
+
+		arch := k8s.GetNodeArchitecture()
+		fmt.Printf("Node architecture: %q\n", arch)
+
+		if arch != IntelArch {
+			return fmt.Errorf(OnlyIntelArch)
+		}
+
+		addressRange, _ := command.Flags().GetString("address-range")
```
Comment (alexellis) [inline cmd/apps/metallb_app.go:41]: Validate this flag by adding a PreRunE handler, you'll see how in the other apps.

File: cmd/apps/metallb_app.go [hunk 3]
```diff
@@ -0,0 +1,111 @@
...
+
+	"github.com/alexellis/arkade/pkg"
+	"github.com/spf13/cobra"
+)
+
+func MakeInstallMetalLB() *cobra.Command {
+	var command = &cobra.Command{
+		Use:          "metallb",
+		Short:        "Install metallb",
+		Long:         `Install metallb for service type:LoadBalancer`,
```
Comment (alexellis) [inline cmd/apps/metallb_app.go:21]: This is taken from the GitHub repo and is more descriptive:

`Install a network load-balancer implementation for Kubernetes using standard routing protocols`

File: cmd/apps/metallb_app.go [hunk 4]
```diff
@@ -0,0 +1,111 @@
...
+		Example:      `arkade install metallb`,
```
Comment (alexellis) [inline cmd/apps/metallb_app.go:22]: Provide an example with an overridden CIDR please

File: cmd/apps/metallb_app.go [hunk 5]
```diff
@@ -0,0 +1,111 @@
...
+			return err
+		}
+
+		err = k8s.Kubectl("apply", "-f",
+			"https://raw.githubusercontent.com/metallb/metallb/v0.9.5/manifests/metallb.yaml")
+		if err != nil {
+			return err
+		}
+
+		token := make([]byte, 32)
```
Comment (alexellis) [inline cmd/apps/metallb_app.go:55]: This should probably be overridden by an optional flag, so users can automate and use a known key.

File: cmd/apps/metallb_app.go [hunk 6]
```diff
@@ -0,0 +1,111 @@
...
+			Name:      "memberlist",
+			Namespace: "metallb-system",
+			SecretData: []types.SecretsData{{
+				Type:  "string-literal",
+				Key:   "secretkey",
+				Value: b64.StdEncoding.EncodeToString(token),
+			}},
+		}
+
+		err = k8s.CreateSecret(secret)
```
Comment (alexellis) [inline cmd/apps/metallb_app.go:69]: Can you inline this?

```
if err := ... ; err != nil {
    return ....
}
```

File: cmd/apps/metallb_app.go [hunk 7]
```diff
@@ -0,0 +1,111 @@
...
+		if err != nil {
+			return fmt.Errorf("Create secret error: %+v", err)
+		}
+
+		configMap := fmt.Sprintf(metalLBConfigMap, addressRange)
+
+		err = k8s.KubectlIn(strings.NewReader(configMap), "apply", "-f", "-")
```
Comment (alexellis) [inline cmd/apps/metallb_app.go:76]: Please also inline this error

File: cmd/apps/metallb_app.go [hunk 8]
```diff
@@ -0,0 +1,111 @@
...
+		fmt.Println(MetalLBInstallMsg)
+
+		return nil
+	}
+
+	return command
+}
+
+const MetalLBInfoMsg = `# Find out more at:
+# https://metallb.universe.tf/
```
Comment (alexellis) [inline cmd/apps/metallb_app.go:90]: You are generating a token, but then don't give the user instructions on how to source or view that token.

A command might help here?

I'd also suggest that you give a command to show them the metallb config map that was generated and applied.

**Review by alexellis — CHANGES_REQUESTED** [review]
Adding review comments, please also update the list of apps in the README by running `arkade get --help`

**Comment by alexellis** [comment]
This requires a rebase, an update to the README, and there are some review comments.

I'd like you to add a note that this is only for ARP mode, so that users do not require BGP, i.e. Equinix Metal customers.

Maybe call the app metallb-arp?

**Comment by alexellis** [comment]
Closed via #512

**Pull request closed without merging**